xcode/Subconscious/Shared/Components/AddressBook/FollowUserView.swift

```diff
@@ -10,68 +10,82 @@ import ObservableStore
 
 struct FollowUserView: View {
     var state: AddressBookModel
-    var send: (AddressBookAction) -> Void
-    
-    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
-    
-    @State var did: String = ""
-    @State var petname: String = ""
-    
-    func validateDid(key: String) -> Did? {
-        Did(key)
-    }
-    
-    func validatePetname(petname: String) -> Petname? {
-        Petname(petname)
+    var form: FollowUserFormModel {
+        get { state.followUserForm }
     }
+    var send: (AddressBookAction) -> Void
     
     var body: some View {
         NavigationStack {
-            Form {
-                Section(header: Text("User To Follow")) {
-                    HStack(alignment: .top) {
-                        Image(systemName: "key")
-                            .foregroundColor(.accentColor)
-                        ValidatedTextField(
-                            placeholder: "DID",
-                            text: $did,
-                            caption: "e.g. did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7",
-                            isValid: validateDid(key: did) != nil || did.count  == 0 // Prevent initial error
-                        )
-                        .lineLimit(1)
-                        .textInputAutocapitalization(.never)
-                        .disableAutocorrection(true)
+            VStack {
+                Form {
+                    Section(header: Text("User To Follow")) {
+                        HStack(alignment: .top) {
+                            Image(systemName: "key")
+                                .foregroundColor(.accentColor)
+                            ValidatedTextField(
+                                placeholder: "DID",
+                                text: Binding(
+                                    get: { form.did.value },
+                                    send: send,
+                                    tag: { v in .didField(.setValue(input: v))}
+                                ),
+                                onFocusChanged: { focused in
+                                    send(.didField(.focusChange(focused: focused)))
+                                },
+                                caption: "e.g. did:key:z6MkmCJAZansQ3p1Qwx6wrF4c64yt2rcM8wMrH5Rh7DGb2K7",
```

Comment:
What does this point to? Anything?

Reply:
I believe it's my testing sphere address, but I might've changed a couple characters. I intend to make a new one before publishing anything other than test data.